javascript/javascript3/week3/products.js

```diff
@@ -0,0 +1,215 @@
+        //Insert into container
+        myProducts.appendChild(myUl);
+    }
+
+    async renderOneProduct(productName) {
+        eraseProducts(myProductSpec)
+        let oneProduct = this.searchProduct(productName)
+        oneProduct = oneProduct[0]
+        insertProduct("h4", oneProduct.name, myProductSpec, "one-product-name")
+        insertProduct("p", productDescription, myProductSpec, "product-description")
+        const url = await getImage(productName).then(data => data.photos[0].src.small)
+        insertImage(url, myProductSpec, "product-img")
+        insertProduct("p", oneProduct.price + " kr", myProductSpec, "one-product-price")
+
+    }
+
+    getUser() {
+        return fetch("https://jsonplaceholder.typicode.com/users/1")
+            .then(res => res.json())
+    }
+
+    renderAllConverted(currency) {
+        //erase previous data
+        eraseProducts(myProducts)
+        //container
+        const myUl = document.createElement("ul");
+        //convert prices of shopping cart
+        this.products.forEach(product => {
+            product.convertToCurrency(currency).then(price => {
+                insertProduct("li", product.name, myUl, "product-name");
+                insertProduct("li", `${price} ${currency}`, myUl, "product-price")
+            })
+        })
+        //get total of converted prices
+        let totalConverted = this.products.map(product => product.convertToCurrency(currency))
+        totalConverted = Promise.all(totalConverted).then(values => values.reduce((a, b) => parseFloat(a) + parseFloat(b))).then(total => insertProduct("li", `${total.toFixed(2)} ${currency}`, myUl, "total"))
+        //append everything
+        myProducts.appendChild(myUl);
+    }
+}
+
+
+//Tests
+const shoppingCart = new ShoppingCart();
+const flatscreen = new Product("flat-screen", 5000);
+const radio = new Product("radio", 500);
+const speakers = new Product("speakers", 300);
+const computer = new Product("computer", 7000);
+const blender = new Product("blender", 900);
+const monitor = new Product("computer-monitor", 1000);
+
+shoppingCart.addProduct(flatscreen);
+shoppingCart.addProduct(radio);
+shoppingCart.addProduct(speakers);
+shoppingCart.addProduct(computer);
+shoppingCart.addProduct(blender);
+shoppingCart.addProduct(monitor);
+console.log(shoppingCart);
+
+shoppingCart.removeProduct(radio);
+console.log(shoppingCart);
+
+const search = shoppingCart.searchProduct("speakers");
+console.log(search);
+shoppingCart.getUser().then(client => shoppingCart.renderProducts(client))
+const total = shoppingCart.getTotal();
+console.log(total);
+
+const plant = new Product("plant", 50);
+const priceDollar = plant.convertToCurrency("USD").then(data => console.log(data))
+
+
+//Autocomplete input
+const autocompleteOptions = shoppingCart.products.map(item => item.name);
+console.log(autocompleteOptions);
+
+var options = autocompleteOptions;
+function autocomplete(inp, arr) {
```

Comment:
same

Reply:
Thank you very much for your comments. I think I managed to improve the functions and modify ones to make them reusable.